kpm/lib/kpm/diagnostic_file.rb

```diff
@@ -0,0 +1,201 @@
+require 'yaml'
+require 'logger'
+require 'tmpdir'
+require 'rubygems'
+require 'zip'
+require 'json'
+require 'fileutils'
+require 'date'
+
+
+module KPM
+
+    class DiagnosticFile
+
+      # Temporary directory
+      TMP_DIR_PREFIX  = 'killbill-diagnostics-'
+      TMP_DIR         = Dir.mktmpdir(TMP_DIR_PREFIX)
+      TMP_LOGS_DIR    = TMP_DIR + File::Separator + 'logs'
+
+      TENANT_FILE     = 'tenant_config'
+      SYSTEM_FILE     = 'system_configuration'
+      ACCOUNT_FILE    = 'account_data'
+
+      TODAY_DATE      = Date.today.strftime('%m-%d-%y')
+      ZIP_FILE        = 'killbill-diagnostics-' + TODAY_DATE + '.zip'
+      ZIP_LOG_FILE    = 'logs.zip'
+
+      @@catalina_base
+
+      def export_data(account_id, config_file = nil, log_dir = nil)
+
+        puts TODAY_DATE
+
+        set_config(config_file)
+
+        tenant_export_file  = get_tenant_config
+        system_export_file  = get_system_config(config_file)
+        account_export_file = get_account_data(config_file, account_id)
+        log_files           = get_log_files(log_dir)
+
+        if File.exist?(account_export_file) && File.exist?(system_export_file) && File.exist?(tenant_export_file)
+
+
+          zip_file_name = TMP_DIR + File::Separator + ZIP_FILE
+
+          Zip::File.open(zip_file_name, Zip::File::CREATE) do |zipFile|
+
+            zipFile.add(TENANT_FILE,  tenant_export_file)
+            zipFile.add(SYSTEM_FILE,  system_export_file)
+            zipFile.add(ACCOUNT_FILE, account_export_file)
+            zipFile.add(ZIP_LOG_FILE, log_files)
+
+          end
+
+          File.delete(tenant_export_file)
+          File.delete(system_export_file)
+          File.delete(account_export_file)
+          File.delete(log_files)
+
+        logger.info "\e[32mDiagnostic data exported under #{zip_file_name} \e[0m"
+
+        else
+          raise Interrupt, 'Account id or configuration file not found'
+        end
+
+      end
+
+      # Private methods
+
+      private
+
+      def get_tenant_config
+
+        logger.info 'Retrieving tenant configuration'
+
+        kb_api_credentials = [get_config('killbill', 'api_key'), get_config('killbill','api_secret')]
+        kb_credentials = [get_config('killbill', 'user'), get_config('killbill','password')]
+        kb_url = 'http://' + get_config('killbill', 'host').to_s + ':' + get_config('killbill','port').to_s
+
+        tenant_config = KPM::TenantConfig.new(kb_api_credentials, kb_credentials, kb_url, logger_suppressed)
+        export_file = tenant_config.export
+
+        final = TMP_DIR + File::Separator + TENANT_FILE
+        FileUtils.cp(export_file, final)
+
+        FileUtils.rm_r(export_file.gsub(export_file.split('/').last, ''))
```

Comment:
This sounds dangerous - could we move the file instead, so we don't have to remove it?